src/blister.c

```diff
@@ -1286,9 +1286,9 @@ static Obj FuncUNITE_BLIST_LIST(Obj self, Obj list, Obj blist, Obj sub)
     UInt                bit;            /* one bit of block                */
     Int                 lenList;        /* logical length of the list      */
     const Obj *         ptrSub;         /* pointer to the sublist          */
-    UInt                lenSub;         /* logical length of sublist       */
-    UInt                i, j, k = 0, l;     /* loop variables                  */
-    long                s, t;           /* elements of a range             */
+    Int                 lenSub;         /* logical length of sublist       */
+    Int                 i, j, k, l;     /* loop variables                  */
+    Int                 s, t;           /* elements of a range             */
```

Comment:
s, t were just of type `long`, so in 64 bit mode, with compilers where long is only 32 bit (basically anywhere except some Microsoft / Windows compilers), there could be overflow here. I didn't add tests for that, though.